api_tests/tests/lightning_routes.ts

```diff
@@ -18,18 +19,29 @@ describe("Lightning routes tests", () => {
             const bodies = (await SwapFactory.newSwap(alice, bob, true))
                 .hanEthereumEtherHalightLightningBitcoin;
 
-            await expect(
-                alice.cnd.createHanEthereumEtherHalightLightningBitcoin(
-                    bodies.alice
-                )
-            ).rejects.toThrow("lightning is not configured.");
-            try {
-                await bob.cnd.createHanEthereumEtherHalightLightningBitcoin(
-                    bodies.bob
-                );
-            } catch (err) {
-                expect(err.status).toBe(400);
-            }
+            const aliceResponse = alice.cnd.createHanEthereumEtherHalightLightningBitcoin(
+                bodies.alice
+            );
+            const bobResponse = bob.cnd.createHanEthereumEtherHalightLightningBitcoin(
+                bodies.bob
+            );
+
+            await expect(aliceResponse).rejects.toMatchObject(
+                new Problem({
+                    status: 400,
+                    title: "lightning is not configured.",
+                    detail:
+                        "lightning ledger is not properly configured, swap involving this ledger are not available.",
```

Comment:
Does this mean this test now breaks if someone edits this string over in `cnd/src/http_api/problem.rs`?  If so, whats the failure going to look like?  Is it going to be obvious that its just caused by the string changing so we don't go chasing shadows?

Reply:
After reading your comments https://github.com/comit-network/comit-rs/pull/2543#discussion_r417147453 I can answer this myself I believe.  If we use static strings for the error and have them at the 'top level' i.e., not returned from deeply nested errors, then this works nicely.

Reply:
The error could be nicer but I think it is acceptable:

![image](https://user-images.githubusercontent.com/5486389/80664815-17ccf380-8adb-11ea-97bc-36dcfd58e104.png)

Reply:
You have to do a manual comparison unfortunately.

Reply:
Actually, now you got me to writing a custom matcher :joy:

Reply:
I tried for a bit but couldn't get anything useful out of it. Will leave as is!